pkg/downstream/suggest.go

```diff
@@ -121,3 +125,77 @@ func iteratePullRequestCommits(ctx context.Context, client *github.Client, org,
 			return client.PullRequests.ListCommits(ctx, org, repo, prNum, o)
 		})
 }
+
+func hasCommit(git utils.GitHelper, req *SuggestRequest, found []string, c *github.Commit) (bool, error) {
+	has_commit := false
+	for _, commit := range found {
+		has, err := compareDiff(git, req, commit, c.GetURL())
+		if err != nil {
+			return false, err
+		}
+		has_commit = has
+	}
+
+	return has_commit, nil
+}
+
+func compareDiff(git utils.GitHelper, req *SuggestRequest, c string, u string) (bool, error) {
+	if len(c) == 0 {
+		return false, nil
+	}
+	remoteDiff, err := pullRemoteDiff(req, u)
+	if err != nil {
+		return false, err
+	}
+	localDiff, err := git.DoOutput("show", "--pretty=format:%n", c)
+	if err != nil {
+		return false, err
+	}
+	remoteDiffLines := strings.Split(strings.TrimSuffix(remoteDiff, "\n"), "\n")
+	localDiffLines := strings.Split(strings.TrimSuffix(localDiff, "\n"), "\n")
+
+	//iterate from back and remove empty lines
+	for i := len(remoteDiffLines) - 1; i >= 0; i-- {
+		if remoteDiffLines[i] == " " {
+			remoteDiffLines = append(remoteDiffLines[:i], remoteDiffLines[i+1:]...)
+		}
+	}
+
+	for i := len(localDiffLines) - 1; i >= 0; i-- {
+		if localDiffLines[i] == " " {
+			localDiffLines = append(localDiffLines[:i], localDiffLines[i+1:]...)
+		}
+	}
+
+	//remote diff sometimes contains an extra blank space as last line
+	if remoteDiffLines[len(remoteDiffLines)-1] == " " {
+		remoteDiffLines = remoteDiffLines[:len(remoteDiffLines)-1]
+	}
+
+	if len(remoteDiffLines) != len(localDiffLines) {
+		return false, nil
+	}
+
+	for i, line := range remoteDiffLines {
+		if line != localDiffLines[i] && !strings.Contains(line, "index") && !strings.Contains(line, "@@") {
```

Comment:
```suggestion
		if line != localDiffLines[i] && !strings.HasPrefix(line, "index") && !strings.HasPrefix(line, "@@") {
```
This can avoid cases where those character are actually part of the diff